File: frontend/cls-budget-web/src/features/accounts/types/account.ts
export interface AccountResponse {
  accountId: number;
  name: string;
  number: string;
  description: string | null;
  balance: number;
  limit: number;
  accountOpenDate: string;
  monthlyPayment: number | null;
  paymentDay: number | null;
  gracePeriod?: number | null;
  graceDay?: number | null;
  phone: string;
  email: string;
  url: string;
  username: string | null;
  notes: string | null;
  isPaidOff: boolean;
  paidOffDate: string | null;
  isCreditCard: boolean | null;
  accountCategoryId: number;
  accountCategoryName?: string | null;
  accountSubCategoryId?: number | null;
  accountSubCategoryName?: string | null;
  /** Purchase APR percent (e.g. 22.99). */
  interestRate: number | null;
  promotionalAnnualPercentageRate?: number | null;
  promotionalRateExpirationDate?: string | null;
  minimumPaymentPercentage?: number | null;
  minimumPaymentFloor?: number | null;
  /** Optional cash advance APR percent (e.g. 28.99). */
  cashOutInterestRate?: number | null;
  /** Optional cash advance fee percent (e.g. 5 = 5%). */
  cashAdvanceFeePercentage?: number | null;
  /** When false, omitted from multi-card payoff analysis. Defaults to true. */
  includeInPayoffAnalysis?: boolean;
}

export interface CreateAccountRequest {
  name: string;
  number: string;
  description?: string | null;
  balance: number;
  limit: number;
  accountOpenDate: string;
  monthlyPayment?: number | null;
  paymentDay?: number | null;
  gracePeriod?: number | null;
  /** Calculated server-side from paymentDay + gracePeriod; optional on write. */
  graceDay?: number | null;
  phone: string;
  email: string;
  url: string;
  username?: string | null;
  password?: string | null;
  notes?: string | null;
  isPaidOff: boolean;
  paidOffDate?: string | null;
  isCreditCard?: boolean | null;
  accountCategoryId: number;
  accountSubCategoryId?: number | null;
  interestRate?: number | null;
  promotionalAnnualPercentageRate?: number | null;
  promotionalRateExpirationDate?: string | null;
  minimumPaymentPercentage?: number | null;
  minimumPaymentFloor?: number | null;
  /** Optional cash advance APR percent (e.g. 28.99). */
  cashOutInterestRate?: number | null;
  /** Optional cash advance fee percent (e.g. 5 = 5%). */
  cashAdvanceFeePercentage?: number | null;
  /** When false, omitted from multi-card payoff analysis. Defaults to true. */
  includeInPayoffAnalysis?: boolean;
}

export type UpdateAccountRequest = CreateAccountRequest;

File: backend/src/CLS.Budget.Domain/Entities/Account.cs
namespace CLS.Budget.Domain.Entities;

public class Account : ITenantOwned
{
    public int AccountId { get; set; }
    public Guid TenantId { get; set; }
    public string Name { get; set; } = null!;
    public string Number { get; set; } = null!;
    public string? Description { get; set; }
    public decimal Balance { get; set; }
    public decimal Limit { get; set; }
    public DateTime AccountOpenDate { get; set; }
    public decimal? MonthlyPayment { get; set; }
    public int? PaymentDay { get; set; }
    /// <summary>Grace period in days after the payment due date.</summary>
    public int? GracePeriod { get; set; }
    /// <summary>Day of month when the grace period ends (PaymentDay + GracePeriod, wrapped 1–31).</summary>
    public int? GraceDay { get; set; }
    public string Phone { get; set; } = null!;
    public string Email { get; set; } = null!;
    public string Url { get; set; } = null!;
        public string? Username { get; set; }
        // NOTE: don’t store plaintext passwords in real apps—hash/salt instead.
        public string? Password { get; set; }
        public string? Notes { get; set; }
        public bool IsPaidOff { get; set; }
        public DateTime? PaidOffDate { get; set; } // (typo fixed from PaidOfDate)
    public bool? IsCreditCard { get; set; }
    public int AccountCategoryId { get; set; }
    public int? AccountSubCategoryId { get; set; }
    public AccountCategory? AccountCategory { get; set; }
    public AccountSubCategory? AccountSubCategory { get; set; }
    public CreditCardDetail? CreditCardDetail { get; set; }
}